ui/src/app/pages/admin/admin-workspace.tsx

```diff
@@ -56,22 +54,24 @@ export class AdminWorkspace extends React.Component<{}, State> {
 
     this.state = {
       googleProject: '',
-      workspace: null,
-      collaborators: [],
-      resources: {},
+      workspaceDetails: {},
+      cloudStorageTraffic: null,
     };
   }
 
-  getFederatedWorkspaceInformation() {
-    workspaceAdminApi().getFederatedWorkspaceDetails(this.state.googleProject).then(
-      response => {
-        this.setState({
-          workspace: response.workspace,
-          collaborators: response.collaborators,
-          resources: response.resources
-        });
-      }
-    );
+  async getFederatedWorkspaceInformation() {
+    this.setState({
+      loadingData: true,
+    });
+
+    // Fire off both requests in parallel
+    const workspaceDetailsPromise = workspaceAdminApi().getFederatedWorkspaceDetails(this.state.googleProject);
+    const cloudStorageTrafficPromise = workspaceAdminApi().getCloudStorageTraffic(this.state.googleProject);
+    // Wait for both promises to complete before updating state.
+    const workspaceDetails = await workspaceDetailsPromise;
+    const cloudStorageTraffic = await cloudStorageTrafficPromise;
```

Comment:
If one of these 503s, this function will short circuit and we will get an infinite spinner. Again as it's an internal page it's not that big a deal, but it could be nice to have it say 'oops it 503'd' instead of just spinning forever when you're looking at it at 3am.